pkg/mixin/provider/delete.go

```diff
@@ -0,0 +1,39 @@
+package mixinprovider
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/deislabs/porter/pkg/mixin"
+	"github.com/pkg/errors"
+)
+
+func (fs *FileSystem) Delete(opts mixin.DeleteOptions) (*mixin.Metadata, error) {
+	if opts.Name != "" {
+		return fs.deleteByName(opts.Name)
+	}
+
+	return nil, errors.New("No mixin name was provided for deletion")
+}
+
+func (fs *FileSystem) deleteByName(name string) (*mixin.Metadata, error) {
+	mixinsDir, err := fs.GetMixinsDir()
+	if err != nil {
+		return nil, err
+	}
+	mixinDir := filepath.Join(mixinsDir, name)
+	exists, _ := fs.FileSystem.Exists(mixinDir)
+	if exists == true {
+		err = fs.FileSystem.RemoveAll(mixinDir)
+		if err != nil {
+			return nil, err
```

Comment:
Anytime we are returning an error from a non-porter library, the custom in Porter is to wrap it using `errors.Wrapf` since the error is going to end up being displayed directly to the user. 

This way we can show the original error, and provide a bit of context about what we were trying to do that caused it. This will help us when people open bugs later and give us back the error message.

Would you please update this to use the errors package?

```suggestion
			return nil, errors.Wrapf(err, "could not remove mixin directory %q", mixinDir)
```